src/components/ApplicationGroup/List/EnvironmentsList.tsx

```diff
@@ -14,125 +14,77 @@
-        history.push(`${match.path}?${queryParams.toString()}`)
-    }
+    const {
+        searchKey,
+        cluster,
+        handleSearch,
+        updateSearchParams,
+        clearFilters,
+        offset,
+        pageSize,
+        changePage,
+        changePageSize,
+    } = urlFilters
 
-    const handleFilterKeyPress = (_searchText: string): void => {
-        setSearchText(_searchText)
-        handleSearch(_searchText)
-    }
+    const filterConfig = useMemo(() => {
+        return { searchKey, cluster, offset, pageSize }
+    }, [searchKey, JSON.stringify(cluster), offset, pageSize])
 
-    const applyFilter = (type: string, list: FilterOption[], selectedAppTab?: string): void => {
-        const queryParams = new URLSearchParams(location.search)
-        const ids = []
-        list.forEach((option) => {
-            if (option.isChecked) {
-                ids.push(option.key)
-            }
-        })
-        if (!ids.length) {
-            queryParams.delete(type)
-        } else {
-            queryParams.set(type, ids.toString())
-        }
-        queryParams.set('offset', '0')
-        history.push(`${match.path}?${queryParams.toString()}`)
-    }
+    const [clusterListLoading, clusterListRes] = useAsync(() => getClusterListMinWithoutAuth())
 
-    const removeFilter = (filter): void => {
-        const queryParams = new URLSearchParams(location.search)
-        const cluster = queryParams.get('cluster')
-        const clusterValues = cluster
-            .toString()
-            .split(',')
-            .map((status) => status)
+    const clusterOptions: SelectPickerOptionType[] = useMemo(() => {
+        return clusterListRes?.result.map((clusterItem) => ({
+            label: clusterItem.cluster_name,
+            value: String(clusterItem.id),
+        })) ?? []
+    }, [clusterListRes])
 
-        const key = clusterValues.filter((value) => value != filter.key)
-        if (key.length) {
-            queryParams.set('cluster', key.toString())
-        } else {
-            queryParams.delete('cluster')
-        }
-        queryParams.set('offset', '0')
-        history.push(`${match.path}?${queryParams.toString()}`)
+    const handleApplyClusterFilter = (clusterOptions: SelectPickerOptionType[]) => {
+        updateSearchParams({ cluster: clusterOptions.map((clusterItem) => String(clusterItem.value)) })
     }
 
-    const removeAllFilters = (): void => {
-        const queryParams = new URLSearchParams(location.search)
-        queryParams.delete('cluster')
-        queryParams.delete('search')
-        queryParams.set('offset', '0')
-        setSearchText('')
-        history.push(`${match.path}?${queryParams.toString()}`)
+    const getFormattedValue = (filterKey: AppGroupUrlFilters , filterValue: string) => {
+        switch (filterKey) {
+            case AppGroupUrlFilters.cluster:
+                return clusterOptions.find((clusterItem) => clusterItem.value === filterValue)?.label
```

Comment:
Are we sure we should that this, can we verify the case of deleted cluster name in url

Reply:
working fine